Simplenote/NoticePresenter.swift

```diff
@@ -0,0 +1,168 @@
+import UIKit
+
+class NoticePresenter {
+
+    // MARK: Properties
+    //
+    var noticeView: NoticeView?
+    private var containerView: PassthruView?
+    private var noticeVariableConstraint: NSLayoutConstraint?
+
+    private var keyboardHeight: CGFloat = .zero
+    private var keyboardFloats: Bool = false
+    private var keyboardNotificationTokens: [Any]?
+
+    private var keyWindow: UIWindow? {
+        return UIApplication.shared.windows.first(where: { $0.isKeyWindow })
+    }
+    private var windowFrame: CGRect {
+        return keyWindow?.frame ?? .zero
+    }
+
+    private var bottomConstraintConstant: CGFloat {
+        let constant = Constants.bottomMarginConstant
+
+        if keyboardFloats || keyboardHeight == .zero {
+            return constant
+        }
+
+        return constant - keyboardHeight
+    }
+
+    // MARK: Lifecycle
+    //
+    deinit {
+        stopListeningToKeyboardNotifications()
+    }
+
+    // MARK: Presenting Methods
+    //
+    func presentNoticeView(_ noticeView: NoticeView, completion: @escaping (Bool) -> Void) {
+        guard let containerView = prepareContainerView() else {
+            return
+        }
+
+        self.noticeView = prepareNoticeView(noticeView, containerView: containerView)
+
+        displayNotificationView(containerView: containerView,
+                                noticeView: noticeView,
+                                completion: completion)
+    }
+
+    private func prepareContainerView() -> PassthruView? {
+        guard let keyWindow = keyWindow else {
+            return nil
+        }
+
+        let containerView = PassthruView(frame: .zero)
+        self.containerView = containerView
+
+        keyWindow.addFillingSubview(containerView)
+
+        return containerView
+    }
+
+    private func prepareNoticeView(_ noticeView: NoticeView, containerView: PassthruView) -> NoticeView {
+        containerView.addSubview(noticeView)
+
+        noticeVariableConstraint = noticeView.topAnchor.constraint(equalTo: containerView.bottomAnchor)
+        noticeVariableConstraint?.isActive = true
+        noticeView.centerXAnchor.constraint(equalTo: containerView.centerXAnchor).isActive = true
+        containerView.layoutIfNeeded()
+
+        return noticeView
+    }
+
+    private func displayNotificationView(containerView: PassthruView, noticeView: NoticeView, completion: @escaping (Bool) -> Void) {
```

Comment:
Maybe `display(view: UIView, in containerView: UIView, completion: @escaping (Bool) -> Void)`?